hazelcast/src/main/java/com/hazelcast/client/impl/ClientSelectors.java

```diff
@@ -0,0 +1,163 @@
+            }
+        };
+    }
+
+    public static ClientSelector none() {
+        return new ClientSelector() {
+            @Override
+            public boolean select(Client client) {
+                return false;
+            }
+
+            @Override
+            public String toString() {
+                return "ClientSelector{none}";
+            }
+        };
+    }
+
+    /**
+     * Regular regex rules are applied. see String.matches
+     *
+     * @param nameMask of the clients that will be selected
+     * @return client selector according to name
+     */
+    public static ClientSelector nameSelector(final String nameMask) {
+        return new ClientSelector() {
+            @Override
+            public boolean select(Client client) {
+                return client.getName().matches(nameMask);
+            }
+
+            @Override
+            public String toString() {
+                return "ClientSelector{nameMask:" + nameMask + " }";
+            }
+        };
+    }
+
+    /**
+     * Works with AddressUtil.mathInterface
+     *
+     * @param ipMask ip mask for the selector
+     * @return client selector according to IP
+     */
+    public static ClientSelector ipSelector(final String ipMask) {
+        return new ClientSelector() {
+            @Override
+            public boolean select(Client client) {
+                return AddressUtil.matchInterface(client.getSocketAddress().getHostName(), ipMask);
+            }
+
+            @Override
+            public String toString() {
+                return "ClientSelector{ipMask:" + ipMask + " }";
+            }
+        };
+    }
+
+    public static ClientSelector or(final ClientSelector... selectors) {
+        return new ClientSelector() {
+            @Override
+            public boolean select(Client client) {
+                for (ClientSelector selector : selectors) {
+                    if (selector.select(client)) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            @Override
+            public String toString() {
+                StringBuilder builder = new StringBuilder();
+                builder.append("ClientSelector{or:");
+                for (ClientSelector selector : selectors) {
+                    builder.append(selector).append(", ");
+                }
+                builder.append("}");
+                return builder.toString();
+            }
+        };
+    }
+
+    /**
+     * Regular regex rules are applied. see String.matches
+     *
+     * @param labelMask of the clients that will be selected
+     * @return client selector according to label
+     */
+    public static ClientSelector labelSelector(final String labelMask) {
+        return new ClientSelector() {
+            @Override
+            public boolean select(Client client) {
+                Set<String> labels = client.getLabels();
```

Comment:
Is it possible that `labels` will be `null`? Same question for `Client#getName`. From `AuthenticationMessageTask` I can see that these fields may be potentially nulls, but I'm not sure. If that's correct, `ClientSelector#select` for labels and instance names may throw NPEs.

Reply:
Good point, we should handle null inside selectors. I will return `false` from selectors when the related item is null.